src/components/StartupViewProvider.ts

```diff
@@ -29,13 +29,18 @@ export class StartupViewProvider implements vscode.WebviewViewProvider {
     webviewView.webview.html = this._getHtmlForWebview(webviewView.webview, this.viewType)
 
     webviewView.webview.onDidReceiveMessage(async (data: { btnType: BUTTON_TYPES }) => {
+      const { shouldShowCliStartUpView, shouldShowWorkspaceView } = await this.checkView()
+      if (data.btnType === BUTTON_TYPES.WORKSPACE) {
+        await vscode.commands.executeCommand('vscode.openFolder')
+        if (!shouldShowWorkspaceView && shouldShowCliStartUpView) {
+          webviewView.webview.html = this._getHtmlForWebview(webviewView.webview, STARTUP_VIEWS.CLI)
```

Comment:
it seems like this logic could be separated from the button action and moved to another method to handle which view should be displayed for the current state, something like
```
webviewView.webview.onDidReceiveMessage(async (data: { btnType: BUTTON_TYPES }) => {
    if (data.btnType === BUTTON_TYPES.WORKSPACE) {
        await vscode.commands.executeCommand('vscode.openFolder')
    }
    refreshStartupView()
})

function refreshStartupView() {
    const { shouldShowCliStartUpView, shouldShowWorkspaceView } = await this.checkView()
    if (shouldShowWorkspaceView) {
        webviewView.webview.html = this._getHtmlForWebview(webviewView.webview, STARTUP_VIEWS.WORKSPACE)
    } else if (shouldShowCliStartUpView) {
        webviewView.webview.html = this._getHtmlForWebview(webviewView.webview, STARTUP_VIEWS.CLI)
    } else {
        await vscode.commands.executeCommand('devcycle-feature-flags.refresh-usages')
    }
}
```